alibi/explainers/integrated_gradients.py

```diff
@@ -543,20 +547,85 @@ def _format_target(target: Union[None, int, list, np.ndarray],
 
     Returns
     -------
-    Formatted target as a list.
+    Formatted target as a np.array.
 
     """
     if target is not None:
         if isinstance(target, int):
-            target = [target for _ in range(nb_samples)]
-        elif isinstance(target, list) or isinstance(target, np.ndarray):
-            target = [t.astype(int) for t in target]
+            target = np.array([target for _ in range(nb_samples)])
+        elif isinstance(target, list):
+            target = np.array(target)
+        elif isinstance(target, np.ndarray):
+            pass
         else:
             raise NotImplementedError
 
     return target
 
 
+def _check_target(output_shape: Tuple,
+                  target: Optional[np.ndarray],
+                  nb_samples: int) -> None:
+    """
+    Parameters
+    ----------
+    output_shape
+        Output shape of the tensorflow model
+    target
+        Target formatted as np array target.
+    nb_samples
+        Number of samples in the batch.
+
+    Returns
+    -------
+    None
+
+    """
+
+    if target is not None:
+
+        if not (target.dtype == int):
+            raise ValueError("Targets must be integers")
+
+        if target.shape[0] != nb_samples:
+            raise ValueError(f"First dimension in target must be egual to nb of samples. "
+                             f"Found target 1st dimension {target.shape[0]}; nb samples: {nb_samples}")
+
+        if len(target.shape) > 2:
+            raise ValueError("Targets must be 1-d or 2-d arrays. In 2-d arrays, each column must contain "
+                             "the target index of the corresponding dimension in the model's output tensor.")
+
+        if len(output_shape) == 1:
```

Comment:
This whole branch is only valid if the (implicit) task is classification and outputs are labels rather than probabilities (hence output shape is 1-d), right? Can you add a comment here to clarify?

Reply:
It's valid when you have the binary classification with  single squash output (probability of class 1 but returns tensor of shape (nb_samples) instead of (nb_samples, 1) ).

Reply:
Hmm remind me, is it valid to pass a model that outputs labels instead of probabilities? Or does the model always have to be probabilistic (in the classification case that is)?

Reply:
Not sure about that.